src/screens/Exchange/Buy.js

```diff
@@ -1,50 +1,124 @@
 // @flow
 
-import React, { useCallback, useState } from "react";
+import React, { useCallback, useState, useEffect } from "react";
 import { View, StyleSheet, Platform } from "react-native";
 import SafeAreaView from "react-native-safe-area-view";
 import { useTranslation } from "react-i18next";
-import { useNavigation, useTheme } from "@react-navigation/native";
-import { usePlatformApp } from "@ledgerhq/live-common/lib/platform/PlatformAppProvider";
+import { useTheme } from "@react-navigation/native";
+import type {
+  Account,
+  AccountLike,
+  CryptoCurrency,
+  TokenCurrency,
+} from "@ledgerhq/live-common/lib/types";
+import { useRampCatalog } from "@ledgerhq/live-common/lib/platform/providers/RampCatalogProvider";
+import { currenciesByMarketcap } from "@ledgerhq/live-common/lib/currencies";
+import { getAccountCurrency } from "@ledgerhq/live-common/lib/account/helpers";
+import { isAccountEmpty } from "@ledgerhq/live-common/lib/account";
+import { useSelector } from "react-redux";
 import extraStatusBarPadding from "../../logic/extraStatusBarPadding";
 import TrackScreen from "../../analytics/TrackScreen";
 import Button from "../../components/Button";
-import { ScreenName } from "../../const";
-import LText from "../../components/LText";
-import BuyOption from "./BuyOption";
-import MoonPay from "../../icons/providers/MoonPay";
-import Coinify from "../../icons/providers/Coinify";
+import { NavigatorName, ScreenName } from "../../const";
+import { useRampCatalogCurrencies } from "./hooks";
+import SelectAccountCurrency from "./SelectAccountCurrency";
+import { track } from "../../analytics";
+import { accountsSelector } from "../../reducers/accounts";
 
 const forceInset = { bottom: "always" };
 
-type Provider = "moonpay" | "coinify" | null;
+type Props = {
+  navigation: any,
+  route: {
+    params: {
+      selectedCurrencyId?: string,
+      accountId?: string,
+      parentId?: string,
+    },
+  },
+};
 
-export default function Buy() {
+export default function OnRamp({ navigation, route }: Props) {
   const { t } = useTranslation();
-  const navigation = useNavigation();
   const { colors } = useTheme();
+  const rampCatalog = useRampCatalog();
+  const allCurrencies = useRampCatalogCurrencies(rampCatalog.value.onRamp);
```

Comment:
If the provided it not loaded at this time, value will be undefined. Make sure the data is loaded by checking value is defined

Reply:
fix uploaded